addons/mass_mailing/static/src/models/mailing_trace.js

```diff
@@ -0,0 +1,62 @@
+/** @odoo-module **/
+
+import { attr, Model } from '@mail/model';
+
+Model({
+    name: 'MailingTrace',
+    fields: {
+        id: attr({ identifying: true }),
+        email: attr(),
+        failure_type: attr(),
+        iconClass: attr({
+            compute() {
+                switch (this.trace_status) {
+                    case 'outgoing':
```

Comment:
we can have a dict `status: css_class` and just return `values[this.trace_status]` (might take 50% less lines of code)